framework/src/main/java/org/checkerframework/common/aliasing/AliasingTransfer.java

```diff
@@ -83,40 +85,62 @@ public TransferResult<CFValue, CFStore> visitAssignment(
      */
     @Override
     protected void processPostconditions(
-            MethodInvocationNode n, CFStore store, ExecutableElement methodElement, Tree tree) {
+            Node n, CFStore store, ExecutableElement methodElement, ExpressionTree tree) {
+        // Process MethodInvocation node only
+        MethodInvocationNode methodInvocationNode = null;
+        ObjectCreationNode objectCreationNode = null;
+        if (n instanceof MethodInvocationNode) {
+            methodInvocationNode = (MethodInvocationNode) n;
+        } else if (n instanceof ObjectCreationNode) {
+            objectCreationNode = (ObjectCreationNode) n;
+        } else {
+            throw new BugInCF(
+                    "Error passing other type of node"
+                            + n.getClass().getSimpleName()
+                            + " into processPostconditions of AliasingTransfer,"
+                            + " expecting MethodInvocationNode or ObjectCreationNode");
+        }
         super.processPostconditions(n, store, methodElement, tree);
-        if (TreeUtils.isEnumSuper(n.getTree())) {
+        if (methodInvocationNode != null && TreeUtils.isEnumSuper(methodInvocationNode.getTree())) {
             // Skipping the init() method for enums.
             return;
         }
-        List<Node> args = n.getArguments();
+        List<Node> args =
+                methodInvocationNode != null
+                        ? methodInvocationNode.getArguments()
+                        : objectCreationNode.getArguments();
         List<? extends VariableElement> params = methodElement.getParameters();
-        assert (args.size() == params.size())
-                : "Number of arguments in "
-                        + "the method call "
-                        + n
-                        + " is different from the"
-                        + " number of parameters for the method declaration: "
-                        + methodElement.getSimpleName();
-
+        // Only check MethodInvocationNode
```

Comment:
Can you explain why? Why don't the args/params have to match for object creations?
Maybe the length needs to be adjusted for one, b/c of enclosing class paramters?

Reply:
Thanks, will update this part.